hazelcast/include/hazelcast/client/serialization/pimpl/SerializationService.h

```diff
@@ -55,30 +57,62 @@ namespace hazelcast {
 
         namespace serialization {
             namespace pimpl {
+                struct HAZELCAST_API ObjectType {
+                    ObjectType() : typeId(0), factoryId(-1), classId(-1) {}
+
+                    ObjectType(int32_t typeId, int32_t factoryId, int32_t classId) : typeId(typeId), factoryId(factoryId),
+                                                                                     classId(classId) {}
+
+                    int32_t typeId;
+                    int32_t factoryId;
+                    int32_t classId;
+
+                    friend std::ostream &operator<<(std::ostream &os, const ObjectType &type) {
+                        os << "typeId: " << type.typeId << " factoryId: " << type.factoryId << " classId: "
+                           << type.classId;
+                        return os;
+                    }
+                };
+
                 class HAZELCAST_API SerializationService {
                 public:
-
                     SerializationService(const SerializationConfig& serializationConfig);
 
                     /**
                     *
                     *  return false if a serializer is already given corresponding to serializerId
                     */
-                    bool registerSerializer(boost::shared_ptr<SerializerBase> serializer);
+                    bool registerSerializer(boost::shared_ptr<StreamSerializer> serializer);
 
                     template<typename T>
                     inline Data toData(const T *object) {
-                        if (NULL == object) {
-                            return Data();
-                        }
-
                         DataOutput output;
 
-                        ObjectDataOutput dataOutput(output, portableContext);
+                        SerializerHolder &serializerHolder = getSerializerHolder();
+                        ObjectDataOutput dataOutput(output, &serializerHolder);
 
                         writeHash<T>(object, output);
 
-                        dataOutput.writeObject<T>(object);
+                        if (NULL == object) {
```

Comment:
`writeObject` seems like doing the same thing. Can we double check if it is the case so that we could remove code duplication ?

Reply:
Ok, reused the ObjectDataOutput::writeObject.